### PR TITLE
Content uploads: Supporting directories for filepath

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: src/katello/client/core/repo.py [hunk 1]
```diff
@@ -558,9 +580,31 @@ def send_content(self, repo_id, upload_id, filepath, chunk=None):
                 self.upload_api.upload_bits(repo_id, upload_id, offset, piece)
                 offset += chunk
 
+    def get_content_data(self, content_type, filepath):
+        unit_key = metadata = None
+        if content_type == "yum":
+            try:
+                unit_key, metadata = generate_rpm_data(filepath)
+            except InvalidRPMError:
+                print _("Cannot parse metadata.json from puppet module '%s'.") % filepath
```
Comment (mptap) [inline src/katello/client/core/repo.py]: you meant the rpm metadata, i guess

File: src/katello/client/core/repo.py [hunk 2]
```diff
@@ -558,9 +580,31 @@ def send_content(self, repo_id, upload_id, filepath, chunk=None):
...
+        elif content_type == "puppet":
+            try:
+                unit_key, metadata = generate_puppet_data(filepath)
+            except ExtractionException:
+                print _("Cannot parse the rpm metadata from package '%s'.") % filepath
```
Comment (mptap) [inline src/katello/client/core/repo.py]: and puppet here

Comment (daviddavis) [inline src/katello/client/core/repo.py]: Good catch!

**Comment by daviddavis** [comment]
This got automatically closed because I merged package-uploads into master. Opening https://github.com/Katello/katello-cli/pull/84 instead.

**Pull request closed without merging**